client/public/index.html

```diff
@@ -27,21 +27,36 @@
             <input type="hidden" value="" name="dataset">
             <i class="mdl-icon-toggle__label material-icons">keyboard_arrow_down</i>
             <label for="dataset" class="mdl-textfield__label">Dataset</label>
-            <ul for="dataset" class="mdl-menu mdl-menu--bottom-left mdl-js-menu" id="dataset-list">
```

Comment:
the `for="dataset"` indicates, that the dropdown will always put the current value into the hidden input field (three lines above) 
try to set this up correctly (see [getmdl-select doku](https://github.com/CreativeIT/getmdl-select#data-val))

Reply:
intelliJ suggested that for is an invalid attribute.